### PR TITLE
Voting system

### PR DESCRIPTION
**Pull request by bobbydigital1986**
Added ability to upvote/downvote posts from the list and show pages.

### CODE REVIEW

File: client/src/services/postVote.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                    if (existingVote?.id === body.newVote?.id) {
+                        body.newVote.vote = body.newVote.vote * 2
+                        return body.newVote
+                    } else {
+                        return existingVote
+                    }
+                })
+                salad.rating = salad.rating + body.newVote.vote
+                return salad
+
```
Comment (ToddLGarrison) [inline client/src/services/postVote.js]: Remove space

**Review by ToddLGarrison — CHANGES_REQUESTED** [review]
One space clean up on postVotes

File: client/src/components/SaladList.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import React, { useState, useEffect } from "react"
 import SaladTile from "./SaladTile"
+import postVote from "../services/postVote"
 
 const SaladList = (props) => {
```
Comment (meehanti23) [inline client/src/components/SaladList.js:6]: remove space

File: client/src/components/SaladList.js [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
 
     const [salads, setSalads] = useState([])
+
```
Comment (meehanti23) [inline client/src/components/SaladList.js:8]: remove space

File: server/src/serializers/saladSerializer.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        
+        if (saladVotes.length > 1) {
+            salad.rating = saladVotes.reduce((a, b) => a.vote + b.vote)
+        } else if (salad.votes.length == 1) {
+            salad.rating = salad.votes[0].vote
+        }
+        else {
+            salad.rating = 0
+        }
+
```
Comment (Jesspinoza) [inline server/src/serializers/saladSerializer.js:14]: line 14 remove space

**Review by bobbydigital1986 — COMMENTED** [review]
Good

**Pull request merged**